pkg/computenode/computenode.go

```diff
@@ -567,29 +567,33 @@ func (n *ComputeNode) BidOnJob(ctx context.Context, shard model.JobShard) error
 run job
 this is a separate method to RunShard because then we can invoke tests on it directly
 */
-func (n *ComputeNode) RunShardExecution(ctx context.Context, shard model.JobShard, resultFolder string) error {
+func (n *ComputeNode) RunShardExecution(ctx context.Context, shard model.JobShard, resultFolder string) (*model.RunCommandResult, error) {
 	// check that we have the executor to run this job
 	e, err := n.getExecutor(ctx, shard.Job.Spec.Engine)
 	if err != nil {
-		return err
+		return &model.RunCommandResult{Error: err}, err
 	}
 	return e.RunShard(ctx, shard, resultFolder)
 }
 
-func (n *ComputeNode) RunShard(ctx context.Context, shard model.JobShard) ([]byte, error) {
+func (n *ComputeNode) RunShard(ctx context.Context, shard model.JobShard) ([]byte, *model.RunCommandResult, error) {
 	shardProposal := []byte{}
+	runOutput := &model.RunCommandResult{}
 
 	verifier, err := n.getVerifier(ctx, shard.Job.Spec.Verifier)
 	if err != nil {
-		return shardProposal, err
+		runOutput.Error = err
+		return shardProposal, runOutput, err
 	}
 	resultFolder, err := verifier.GetShardResultPath(ctx, shard)
 	if err != nil {
-		return shardProposal, err
+		runOutput.Error = err
+		return shardProposal, runOutput, err
```

Comment:
I am not a big fan that `RunShard` now returns two types of errors that callers would have to check for, even though _most_ of the time the errors are identical. I know that I suggested over slack to keep returning an `err` value, but I didn't expect that we would still have to encapsulate the error inside `runOutput` instance as well.

While I am still new to golang, but comparing to Java, I am assuming the best practice is to return a dedicate `err` value as callers would know they need to handle it compared to encapsulating the error in another type. However, if we believe in this specific case that encapsulating the error inside `runOutput` would make things more readable and manageable, then I am happy with that as long as we drop the `err` value and only return an error inside `runOutput`

Reply:
Yeah it makes sense given that `runOutput.Error` is just for reference and callers wouldn't need to handle it. Maybe using `errorMsg` of type `string` would avoid future confusion instead of `error` type, but both are fine. Thanks for the clarification :)